docs/src/user_docs/assembly/code_organization.md

```diff
@@ -3,35 +3,28 @@ A Miden assembly program is just a sequence of instructions each describing a sp
 
 In turn, Miden assembly instructions are just keywords which can be parameterized by zero or more parameters. The notation for specifying parameters is *keyword.param1.param2* - i.e., the parameters are separated by periods. For example, `push.123` instruction denotes a `push` operation which is parameterized by value `123`.
 
-Currently, Miden assembly provides two types of code organization blocks: *programs* and *procedures*. In the future, additional blocks such as modules and functions will be added.
+Miden assembly programs are organized into procedures. Procedures, in turn, can be grouped into modules.
 
-### Programs
-A *program* block is used to define an executable program. A program must start with a `begin` instruction and terminate with an `end` instruction. For example:
-```
-begin
-    <instructions>
-end
-```
-When Miden assembly code is executed, the execution starts at the first instruction following the `begin` instruction of the program. A program block is expected to be the last block in an assembly file and can be followed only by comment blocks.
+Currently, Miden assembly provides two types of code organization blocks: *programs* and *procedures*. In the future, additional blocks such as modules and functions will be added.
 
 ### Procedures
-A *procedure* block is used to define a frequently-used sequence of instructions. A procedure must start with a `proc.<label>.<number of locals>` instruction and terminate with an `end` instruction. For example:
+A *procedure* can be used to define a frequently-used sequence of instructions. A procedure must start with a `proc.<label>.<number of locals>` instruction and terminate with an `end` instruction. For example:
```

Comment:
"frequently-used sequence of instructions" is correct, but this sentence could be clearer. What we're trying to say is that you can take a frequently-used sequence of instructions, encapsulate them into a single procedure, and then execute them by calling the procedure.

Maybe it would be clearer to replace "define" with "encapsulate" and then specify that we define a label for it so that it can later be executed by referencing its label

Reply:
Updated!